lib/front_desk.rb

```diff
@@ -0,0 +1,187 @@
+require_relative "reservation"
+require_relative "block_room"
+
+module Hotel
+  class FrontDesk
+    # attr_reader :all_rooms, :reservations_record, :block_reservations_only, :reserved_rooms_in_blocks
+    attr_accessor :all_rooms, :reservations_record, :block_reservations_only, :reserved_rooms_in_blocks
+
+    def initialize
+      @all_rooms = [*(1..20)] # => makes array with elements from 1 to 20
+
+      @reservations_record = Array.new # => stores all reservation objects
+
+      @block_reservations_only = Array.new # Stores all block reservations
+
+      @reserved_rooms_in_blocks = Array.new # stores the reservations of rooms reserved in blocks
+    end
+
+    def all_rooms_array
+      return @all_rooms
+    end
+
+    def reservations_record_array
+      return @reservations_record
+    end #
+
+    def block_reservations_array
+      return @block_reservations_only
+    end
+
+    def reserved_rooms_in_blocks_array
+      return @reserved_rooms_in_blocks
+    end
+
+    def reservation_template(new_booking_ref, new_number_of_rooms, new_first_room_number, new_check_in, new_check_out)
+      booking = Hotel::Reservation.new(
+        booking_ref: new_booking_ref,
+        number_of_rooms: new_number_of_rooms,
+        first_room_number: new_first_room_number,
+        check_in: new_check_in,
+        check_out: new_check_out,
+        total_cost: @total_cost,
+        room_blocks: @room_blocks,
+      )
```

Comment:
Here, when you're calling Reservation.new, you are saying that the value that should go into the parameter `total_cost` should be whatever `@total_cost` is... but `@total_cost` doesn't have any value stored on it, ever! You never call `@total_cost = ...` ever in your code, so it will _always_ be `nil`.

ame with `@room_blocks`. `@room_blocks` is the name of the variable that this instance of `FrontDesk` has stored, but this instance of `FrontDesk` will never change that value, so it's always `nil`. What value were you expecting to be on `@room_blocks`?